lib/rules/array-callback-return.js

```diff
@@ -211,6 +215,14 @@ module.exports = {
                 if (options.checkForEach && node.type === "ArrowFunctionExpression" && node.expression) {
                     messageId = "expectedNoReturnValue";
                 }
+
+                if (options.checkForEach &&
+                    options.allowVoid &&
+                    node.type === "ArrowFunctionExpression" &&
+                    node.body.type === "UnaryExpression" &&
+                    node.body.operator === "void") {
+                    return;
+                }
```

Comment:
I think we can make this a bit easier to follow by pulling the common checks up a level.

```suggestion
                if (options.checkForEach && node.type === "ArrowFunctionExpression") {

                    if (node.expression) {
                        messageId = "expectedNoReturnValue";
                    }
    
                    if (options.allowVoid &&
                        node.body.type === "UnaryExpression" &&
                        node.body.operator === "void") {
                        return;
                    }
                }
```

Reply:
Done!
I think `node.expression` is `true`  for both the conditions